torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -846,7 +862,17 @@ def state_dict(self, *args, **kwargs):
         if torch.cuda.is_available():
             torch.cuda.synchronize()
         if self._state_dict_type == StateDictType.FULL_STATE_DICT:
-            return super().state_dict(*args, **kwargs)
+            self._lazy_init()
```

Comment:
Can we move this outside of the `if` to ensure all different state_dict call `_lazy_init`?